modules/View.js

```diff
@@ -317,38 +420,57 @@
      */
     _detached: _.noop,
 
-        /**
-     * Before any DOM rendering is done, this method is called and removes any
-     * custom plugins including events that attached to the existing elements.
-     * This method can be overwritten as usual OR extended using <baseClass>.prototype.plug.apply(this, arguments);
+    /**
+     * Hook during render that is invoked before any DOM rendering is performed.
+     * This method can be overwritten as usual OR extended using <baseClass>.prototype.prerender.apply(this, arguments);
      * NOTE: if you require the view to be detached from the DOM, consider using _detach callback
-     * @method unplug
+     * @method prerender
      */
-    unplug: _.noop,
+    prerender: _.noop,
 
     /**
-     * After all DOM rendering is done, this method is called and attaches any
-     * custom plugins to the existing elements.  This method can be overwritten
-     * as usual OR extended using <baseClass>.prototype.plug.apply(this, arguments);
+     * Hook to inject in (or transition in) tracked views. Called after all DOM rendering is done so injection sites should be available.
+     * This method can be overwritten  as usual OR extended using <baseClass>.prototype.attachTrackedViews.apply(this, arguments);
+     * @method attachTrackedViews
+     */
+    attachTrackedViews: _.noop,
+
+    /**
+     * Hook during render that is invoked after all DOM rendering is done and tracked views attached.
+     * This method can be overwritten as usual OR extended using <baseClass>.prototype.postrender.apply(this, arguments);
      * NOTE: if you require the view to be attached to the DOM, consider using _attach callback
-     * @method plug
+     * @method postrender
      */
-    plug: _.noop,
+    postrender: _.noop,
 
     /**
-     * Gets the hash from id to views of the correct views given the options.
-     * @param [options={}] {Object}  Optional options.
-     *   @param [options.shared=false] {Boolean} The view is a shared view instead of a child view
-     *                                           (shared views are not disposed when the parent is disposed)
-     * @method __getTrackedViewsHash
+     * Override to provide your own transition out logic. Default logic is to just detach from the page.
+     * @method transitionOut
+     * @param done {Function} callback to be invoked when the transitions is complete *MUST BE CALLED*
```

Comment:
the callback "done" must be invoked at sometime during the transitionOut method. Very much how Jasmine tests must invoke done()